src/llama-kv-cache-unified.cpp

```diff
@@ -613,17 +624,25 @@ int32_t llama_kv_cache_unified::find_slot(const llama_ubatch & ubatch) const {
         }
     }
 
+    uint32_t n_found  = 0;
     uint32_t n_tested = 0;
 
+    const uint32_t n_test = cont ? n_tokens : 1;
+
+    slot_info res;
+
+    res.idxs.resize(n_tokens);
```

Comment:
Replacing this with `reserve` and using `push_back` would remove the possibility of an OOB write.